Doberman/AlarmNode.py

```diff
@@ -0,0 +1,124 @@
+import Doberman
+
+
+class AlarmNode(Doberman.Node):
+    """
+    An empty base class to handle database access
+    """
+    def setup(self, **kwargs):
+        super().setup(**kwargs)
+        self.description = kwargs['description']
+        self.sensor = kwargs['sensor']
+        self._log_alarm = kwargs['log_alarm']
+        self.escalation_config = kwargs['escalation_config']
+        self.escalation_level = 0
+        self.base_level = kwargs['base_level']
+        self.auto_silence_duration = kwargs.get('silence_duration', 5*60)
+        self.alarm_start = None
+        self.hash = None
+
+    def escalate(self):
+        """
+        Do we escalate? This function decides this
+        """
+        if self.alarm_start is None:
+            # not sure how we got here, but there's nothing to escalate
+            return
```

Comment:
Log this?

Reply:
We could, but I don't think it's possible for `escalate` to get called if `alarm_start` isn't set. `escalate` is only called in one place (L55), which happens after setting the property.